root/user/UserProfile.js

```diff
@@ -0,0 +1,515 @@
+            l('0')
+          )}
+          {$c.user_exists && !viewingOwnProfile ? (
+            <>
+              {' '}
+              {bracketed(
+                subscribed ? (
+                  <a href={`/account/subscriptions/editor/remove?id=${encodeURIComponent(user.name)}`}>
+                    {l('unsubscribe')}
+                  </a>
+                ) : (
+                  <a href={`account/subscriptions/editor/add?id=${encodeURIComponent(user.name)}`}>
+                    {l('subscribe')}
+                  </a>
+                ),
+              )}
+            </>
+          ) : null}
+        </UserProfileProperty>
+
+        {user.biography ? (
+          <UserProfileProperty className="biography" name={l('Bio:')}>
+            {showBioAndURL ? (
+              <div dangerouslySetInnerHTML={{__html: user.biography}} />
+            ) : (
+              <div className="deleted">
+                {exp.l(
+                  `This content is hidden to prevent spam.
+                  To view it, please {url|log in}.`,
+                  {url: '/account/login'},
+                )}
+              </div>
+            )}
+          </UserProfileProperty>
+        ) : null}
+
+        {user.languages && user.languages.length > 0 ? (
+          <UserProfileProperty name={l('Languages:')}>
+            <ul className="inline">
+              {user.languages.map(language => (
+                <li key={language.language.id}>
+                  {`${l_languages(language.language.name)} (${l(language.fluency)})`}
+                </li>
+              ))}
+            </ul>
+          </UserProfileProperty>
+        ) : null}
+      </table>
+    </>
+  );
+});
+
+type EditStatsT = {|
+  +accepted_auto_count: number,
+  +accepted_count: number,
+  +cancelled_count: number,
+  +failed_count: number,
+  +last_day_count: number,
+  +open_count: number,
+  +rejected_count: number,
+|};
+
+type VoteStatsT = Array<{|
+  +all: {
+    +count: number,
+    +percentage: number,
+  },
+  +name: string,
+  +recent: {
+    +count: number,
+    +percentage: number,
+  },
+|}>;
+
+type UserProfileStatisticsProps = {|
+  +editStats: EditStatsT,
+  +user: EditorT,
+  +votes: VoteStatsT,
+|};
+
+const UserProfileStatistics = ({
+  editStats,
+  user,
+  votes,
+}: UserProfileStatisticsProps) => {
+  const voteTotals = votes.pop();
+
+  return (
+    <>
+      <h2>{l('Statistics')}</h2>
+
+      <table className="statistics">
+        <thead>
+          <tr>
+            <th colSpan="2">
+              {exp.l(
+                'Edits ({view_url|view})',
+                {view_url: `/user/${encodeURIComponent(user.name)}/edits/all`},
+              )}
+            </th>
+          </tr>
+        </thead>
+
+        <tbody>
+          <UserProfileProperty name={lp('Accepted', 'edit descriptor')}>
+            {exp.l(
+              '{count} ({view_url|view})',
+              {
+                count: Number(editStats.accepted_count).toLocaleString(),
```

Comment:
Using `formatCount` from root/statistics/utilities.js seems like a good idea since it passes the current UI language to `toLocaleString` (rather than defaulting, I assume, to the browser's language).

Reply:
Done, but we should probably consider moving the function out of statistics/utilities and using it in PaginatedResults and TagCloud too then, since they also use toLocaleString().

Wondering also whether to use formatPercentage (from the same file) for percentages here which are currently l('{percentage}%').